lib/api/src/tests/stories.test.js

```diff
@@ -417,4 +451,19 @@ describe('stories API', () => {
       expect(navigate).toHaveBeenCalledWith('/story/a--1');
     });
   });
+
+  describe('showSettingsPage', () => {
+    it('navigates to the specified settings page', () => {
+      const navigate = jest.fn();
+      const store = createMockStore();
+      const {
+        api: { showSettingsPage },
+        state,
+      } = initStories({ store, navigate, viewMode: 'story', storyId: 'a--2' });
+
+      showSettingsPage('about');
+      const { viewMode } = store.getState(state);
+      expect(viewMode).toEqual('settings');
```

Comment:
I'm sorry but how to write a test when `setState` callback is called with `navigate`? I've tried some async test, but nothing worked well. If `showSettingsPage()` itself returns a promise or a callback style, I could do it but It just [calls](https://github.com/storybookjs/storybook/pull/7241/files#diff-9ebe848d7050f56e9d0b9ffee61ba684R317) `store.setState` and callback is passed here. I wanted to write something like `expect(navigate).toHaveBeenCalledWith('/settings/about')`